### ISSUE TITLE
How to relay mpegts ?

### ISSUE THREAD

**Issue by jhdscript** [illuspas/Node-Media-Server#62]
hello, i m trying to relay http mepgts flux but it donna work.

Here is my sample code:

var NodeMediaServer = require('node-media-server');

var LOG_TYPE_NOTHING = 0;
var LOG_TYPE_ERROR = 1;
var LOG_TYPE_INFO = 2;
var LOG_TYPE_DEBUG = 3;

var config = {
  logType: LOG_TYPE_DEBUG,
  rtmp: {
    port: 1935,
    chunk_size: 60000,
    gop_cache: true,
    ping: 60,
    ping_timeout: 30
  },
  http: {
    port: 8000,
    allow_origin: '*'
  },
  relay: {
    ffmpeg: "C:\\ffmpeg\\bin\\ffmpeg.exe",
    tasks: [
     {
      app: 'iptv',
      mode: 'static',
      edge: 'http://iprtttv.org:8000/live/resc2/mIzKJa6M/106.ts',
      name: 'test'
    }]
  }
};

var nms = new NodeMediaServer(config);
nms.run();

Maybe have you an idea ? Help appreciated. Regards

**Comment by illuspas** [comment]
On the Windows platform, ffmpeg paths require double slashes.
```
ffmpeg: "C:\\ffmpeg\\bin\\ffmpeg.exe",
```

**Comment by jhdscript** [comment]
Yes but the problem is my source http is mpeg-ts, so i need to pass some arguments to ffmpeg.

When I send directly using ffmpeg with this command it works.

.\ffmpeg.exe -i http://iprtttv.org:8000/live/resc2/mIzKJa6M/106.ts -c:v libx264 -preset superfast -tune zerolatency -c:a aac -ar 44100 -f flv rtmp://localhost/live/STREAM_NAME

But when I try with relay it doesn t work. Any idea to make it work good ?

**Comment by jhdscript** [comment]
Ok i find a pist. I feedback in some hours

**Comment by illuspas** [comment]
From your parameters, your ts stream needs to be transcode, right?
Relay module is currently only used for stream transfer and no transcoding. 
This requires that the video and audio are encoded by h.264+aac.
If you need to, refer to the trans module to modify ffmpeg parameters.

**Comment by jhdscript** [comment]
Stream provides from an iptv provider. I don t think i need to transcode, i only need to relay to multiple client.

If you need to, refer to the trans module to modify ffmpeg parameters => Have you a sample for remote stream source ?

**Comment by jhdscript** [comment]
I want to use ffmpeg only to relay the video and audio as original

**Comment by CameronSima** [comment]
Did you find a solution for this?
